account/watcher/watcher.go

```diff
@@ -0,0 +1,103 @@
+package watcher
+
+import (
+	"sync"
+
+	"github.com/btcsuite/btcd/btcec"
+)
+
+// expiryWatcher implements the ExpiryWatcher interface
+type expiryWatcher struct {
+	handlers EventHandler
+
+	// bestHeight is the height we believe the current chain is at.
+	bestHeight uint32
+
+	// expirations keeps track of the current accounts we're
+	// watching expirations for.
+	expirations map[[33]byte]uint32
+
+	// expirationsPerHeight keeps track of all registered accounts
+	// that expire at a certain height.
+	expirationsPerHeight map[uint32][]*btcec.PublicKey
+
+	expirationsMtx sync.Mutex
+}
+
+// NewExpiryWatcher instantiates a new ExpiryWatcher.
+func NewExpiryWatcher(handlers EventHandler) *expiryWatcher { // nolint:golint
+	return &expiryWatcher{
+		handlers:             handlers,
+		expirations:          make(map[[33]byte]uint32),
+		expirationsPerHeight: make(map[uint32][]*btcec.PublicKey),
+	}
+}
+
+// NewBlock updates the current bestHeight.
+func (w *expiryWatcher) NewBlock(bestHeight uint32) {
+	w.bestHeight = bestHeight
```

Comment:
Since expiry happens when the height is changed, it'd be possible to simplify the interface a bit and do the height change and the expiry handling atomically under the same lock. By only expiring the current block if two blocks come right after each other in succession we may miss an expiration (almost no chance in practice but still).